src/pages/index.js

```diff
@@ -59,54 +32,126 @@ const config = {
   inputErrorClass: "modal__input_type_error",
   errorClass: "modal__error_visible",
 };
+const profileButton = document.querySelector(".profile__button");
+
+const profileAvatarModal = document.querySelector("#profile-avatar-modal");
+const profileImage = document.querySelector(".profile__image");
+const profileAvatarModalForm = profileAvatarModal.querySelector(".modal__form");
+
+const userInfo = new UserInfo(".profile__title", ".profile__description");
+const cardSection = new Section(renderCard, cardListEl);
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f1dab6ed-e5ba-44ef-bef1-7369d7e9bf0d",
+    "Content-Type": "application/json",
+  },
+});
+
+api.getUserInfo().then((res) => {
+  userInfo.setUserInfo(res);
+});
+api.getInitialCards().then((res) => {
+  cardSection.renderItems(res);
+});
+api.getUserInfo().then((res) => {
+  profileImage.src = res.avatar;
+});
+
+const cardDeleteModal = new PopupQuestion("#delete-card-modal");
 
 const editFormValidator = new FormValidator(config, profileEditForm);
 editFormValidator.enableValidation();
 const addFormValidator = new FormValidator(config, addCardFormElement);
 addFormValidator.enableValidation();
+const profileAvatarFormValidator = new FormValidator(
+  config,
+  profileAvatarModalForm
+);
 
+profileAvatarFormValidator.enableValidation();
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+const cardPopup = new PopupWithForm(
+  "#add-card-modal",
+  // handleAddCardFormSubmit
+  (link, submitButton) => {
+    api.addCard(link, submitButton).then((res) => {
+      cardSection.addItem(res);
+    });
+    addFormValidator.toggleButtonState();
+    cardPopup.close();
+  }
+);
 cardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const profilePopup = new PopupWithForm(
+  "#profile-avatar-modal",
+  (url, submitButton) => {
+    // handles the avatar picture submit
+    api.setUserAvatar(url, submitButton).then((res) => {
+      profileImage.src = res.avatar;
+    });
+
+    profileAvatarFormValidator.toggleButtonState();
+    profilePopup.close();
+  }
+);
+profilePopup.setEventListeners();
+
 const userInfoPopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileFormSubmit
+  (values, submitButton) => {
+    userInfo.setUserInfo(values);
+    api.setUserInfo(values, submitButton);
+    userInfoPopup.close();
+  }
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```